verifiers/internal/gcb/verifier.go

```diff
@@ -0,0 +1,37 @@
+package gha
+
+import (
+	"context"
+	"strings"
+
+	"github.com/slsa-framework/slsa-verifier/options"
+	register "github.com/slsa-framework/slsa-verifier/register"
+)
+
+const VerifierName = "GCB"
+
+//nolint:gochecknoinits
+func init() {
+	register.RegisterVerifier(VerifierName, GCBVerifierNew())
+}
+
+type GCBVerifier struct{}
+
+func GCBVerifierNew() *GCBVerifier {
+	return &GCBVerifier{}
+}
+
+// Match a BuilderID.
+func (v *GCBVerifier) Match(builderID string) bool {
+	// This verifier only supports the GCB builders.
+	return strings.HasPrefix(builderID, "https://cloudbuild.googleapis.com/GoogleHostedWorker@")
+}
+
+// Verify provenance.
+func (v *GCBVerifier) Verify(ctx context.Context,
+	provenance []byte, artifactHash string,
+	provenanceOpts *options.ProvenanceOpts,
+	builderOpts *options.BuilderOpts,
```

Comment:
I believe GCB v2 provenance is signed with a configured key in the workers.

Would the option for specifying public keys exist in `BuilderOpts`?

Reply:
honest answer is TBD :) I agree we need to think of supporting keys. I'm always worried about that because it's dangerous. So this s TBD. I think it's do-able thru a new `BuilderOpts` I you say.